metaflow/plugins/aip/aip.py

```diff
@@ -298,6 +298,16 @@ def _create_workflow_yaml(
 
             # Service account is added through webhooks.
             workflow["spec"].pop("serviceAccountName", None)
+
+            # Parameters with no defaults need to be added to WorkflowTemplates.
+            # This allows the parameters to be supplied by Workflow that reference the WorkflowTemplate.
+            workflow["spec"]["arguments"]["parameters"].extend(
+                [
+                    dict(name=param)
+                    for param in self.flow._get_parameters()
+                    if param not in flow_parameters
+                ]
+            )
```

Comment:
this may not work:
```python
File "/home/zservice/metaflow/metaflow/task.py", line 548, in run_step
    self._exec_step_function(step_func)
  File "/home/zservice/metaflow/metaflow/task.py", line 53, in _exec_step_function
    step_function()
  File "/home/zservice/metaflow/metaflow/plugins/aip/tests/flows/flow_triggering_flow.py", line 89, in start
    self.submit_template(path)
  File "/home/zservice/metaflow/metaflow/plugins/aip/tests/flows/flow_triggering_flow.py", line 198, in submit_template
    subprocess.run(
  File "/usr/local/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['argo', 'template', '-n', 'metaflow-integration-testing-internal', 'create', '/tmp/wfdsk-ftf-test-argo-d64fe9ff-f1d7-4c7c-8551-c45c3c600063-0.yaml']' returned non-zero exit status 1.
```
see:
 * https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/sdks/zillow-metaflow/-/jobs/42842006
 * https://argo-server.int.dev-k8s.zg-aip.net/argo-ui/archived-workflows/metaflow-integration-testing-internal/d64fe9ff-f1d7-4c7c-8551-c45c3c600063

Reply:
note: it'll also be important to ensure it's tested with a workflow compiled with an exit handler to ensure we don't hit https://github.com/argoproj/argo-workflows/issues/6036